api/users/serializers.py

```diff
@@ -34,6 +35,27 @@ def to_representation(self, value):
         return relationship_links
 
 
+class SocialField(ser.DictField):
+    def __init__(self, min_version, **kwargs):
+        super(SocialField, self).__init__(**kwargs)
+        self.min_version = min_version
+        self.help_text = 'This field will change data formats after version {}'.format(self.min_version)
+
+    def to_representation(self, value):
+        old_social_string_fields = ['twitter', 'github', 'linkedIn']
+        request = self.context.get('request')
+        show_old_format = request and is_deprecated(request.version, self.min_version) and request.method == 'GET'
```

Comment:
Ah! So this was because I wasn't exactly sure if versioning was for other operations other than get, and by extension didn't know if we should support patches/puts for data that was in the legacy format. If we indeed are going to support a patch to the social field in the old format, as well as the new format if they're using a current version of the API, that will be a little more work as the jsonschema isn't set up to validate based on version of the API. 

Lemmie know and I can make that happen!

Reply:
Oh right--non-GET requests were disallowed in older versions. So maybe the `request.method == 'GET'` is just redundant? Can we remove it?

Reply:
The reason I added the request.method check was because updates actually secretly **were** allowed in earlier versions -- this feature was actually present already and this PR was just intended to clean it up/make it work the way that's expected. 

This was added primarily to make the [tests in this method](https://github.com/CenterForOpenScience/osf.io/blob/develop/api_tests/users/views/test_user_detail.py#L384) pass while still not specifying the version of the API, cause they were using version 2.0.  I figured since this was a stealth feature before anyhow, it'd be ok to require "proper form" for `PUT`/`POST` requests, while still returning data in the way that's expected for `GET` requests.